src/xunit.v3.core/CollectionDefinitionAttribute.cs

```diff
@@ -13,6 +13,15 @@ namespace Xunit;
 [AttributeUsage(AttributeTargets.Class, AllowMultiple = false, Inherited = false)]
 public sealed class CollectionDefinitionAttribute : Attribute
 {
+	/// <summary>
+	/// Initializes a new instance of the <see cref="CollectionDefinitionAttribute" /> class.
+	/// Only use this constructor when collection references by test classes use the generic
+	/// <see cref="CollectionAttribute{TCollectionDefinition}"/> attribute or refer to the
+	/// fixture class using typeof(CollectionDefinition).
+	/// </summary>
+	public CollectionDefinitionAttribute()
```

Comment:
I think with the understanding that `[Collection(type)]` and `[Collection<T>]` meaning we don't need `[CollectionDefinition]`, we probably don't want to add this constructor. We can leave this class only for definitions discovered by name.

Reply:
I would agree, except `CollectionDefinitionAttrtibute` has the parallelism property that has no other way of being specified. Without the default constructor, one would always have to specify a name to also specify the other parameter.

Reply:
Said differently, if `CollectionDefinitionAttribute` had no fields other than name/type, I'd 100% agree, but there is one itty-bitty teeny-weeny boolean value still there that has no other way to be set.

Reply:
Good point. It can become optional then.